lib/versioner/simple.go

```diff
@@ -63,6 +72,76 @@ func (v simple) Archive(filePath string) error {
 	return nil
 }
 
+func (v *simple) serve(ctx context.Context) {
+	l.Debugln(v, "starting")
+	defer l.Debugln(v, "stopping")
+
+	// Do the first cleanup one minute after startup.
+	timer := time.NewTimer(time.Minute)
+	defer timer.Stop()
+
+	for {
+		select {
+		case <-ctx.Done():
+			return
+
+		case <-timer.C:
+			if v.cleanOutDays > 0 {
+				if err := v.cleanOutArchive(); err != nil {
+					l.Infoln("Cleaning simple:", err)
+				}
+			}
+
+			// Cleanups once a day should be enough.
+			timer.Reset(24 * time.Hour)
+		}
+	}
+}
+
+func (v *simple) String() string {
+	return fmt.Sprintf("simple@%p", v)
+}
+
+func (v *simple) cleanOutArchive() error {
+	if _, err := v.versionsFs.Lstat("."); fs.IsNotExist(err) {
+		return nil
+	}
+
+	cutoff := time.Now().Add(time.Duration(-24*v.cleanOutDays) * time.Hour)
+	dirTracker := make(emptyDirTracker)
+
+	walkFn := func(path string, info fs.FileInfo, err error) error {
+		if err != nil {
+			return err
+		}
+
+		if info.IsDir() && !info.IsSymlink() {
+			dirTracker.addDir(path)
+			return nil
+		}
+		
+		versionTime, err := time.ParseInLocation(TimeFormat, extractTag(path), time.Local)
```

Comment:
Check the error return; if there is an error we should just skip the file (`return nil`)